src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "lazy_tensor.hpp"
+
+using ov::npuw::weights::ConcatMeta;
+using ov::npuw::weights::LazyTensor;
+using ov::npuw::weights::Transform;
+using ov::npuw::weights::TransformType;
+
+std::size_t LazyTensor::Hash::operator()(const LazyTensor& lt) const {
+    std::size_t seed = std::hash<void*>()(lt.m_orig_data) + 0x9e3779b9;
+    seed ^= std::hash<std::string>()(lt.m_orig_shape.to_string()) + 0x9e3779b9;
+    seed ^= std::hash<std::string>()(lt.m_orig_type.to_string()) + 0x9e3779b9;
```

Comment:
Seen this before but where these magic numbers come from again??

Reply:
https://softwareengineering.stackexchange.com/questions/402542/where-do-magic-hashing-constants-like-0x9e3779b9-and-0x9e3779b1-come-from